pages/CitationStyleLanguageHandler.inc.php

```diff
@@ -105,54 +108,48 @@ public function _setupRequest($args, $request)
+        $this->submission = Repo::submission()->get($userVars['submissionId']);
+        $this->issue = $userVars['issueId'] ? Repo::issue()->get($userVars['issueId']) : null;
+        $this->submissionNoun = $userVars['submissionNoun'];
 
-        if (!$this->article) {
+        if (!$this->submission) {
             $request->getDispatcher()->handle404();
         }
 
-        $this->publication = !empty($userVars['publicationId'])
-            ? Repo::publication()->get($userVars['publicationId'])
-            : $this->article->getCurrentPublication();
-
-        if ($this->article) {
-            // Support OJS 3.1.x and 3.2
-            $issueId = method_exists($this->article, 'getCurrentPublication') ? $this->article->getCurrentPublication()->getData('issueId') : $this->article->getIssueId();
-            $issue = Repo::issue()->get($issueId);
-            $issue = $issue->getJournalId() == $context->getId() ? $issue : null;
-            $this->issue = $issue;
-        }
+        $applicationName = Application::get()->getName();
 
         // Disallow access to unpublished submissions, unless the user is a
         // journal manager or an assigned subeditor or assistant. This ensures the
-        // article preview will work for those who can see it.
-        if (!$this->issue || !$this->issue->getPublished() || $this->article->getStatus() != PKPSubmission::STATUS_PUBLISHED) {
-            $userCanAccess = false;
-
-            if ($user && $user->hasRole([Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT], $context->getId())) {
-                $isAssigned = false;
-                $userGroupDao = DAORegistry::getDAO('UserGroupDAO');
-                $stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
-                $assignments = $stageAssignmentDao->getBySubmissionAndStageId($this->article->getId());
-                foreach ($assignments as $assignment) {
-                    if ($assignment->getUser()->getId() !== $user->getId()) {
-                        continue;
-                    }
-                    $userGroup = $userGroupDao->getById($assignment->getUserGroupId($context->getId()));
-                    if (in_array($userGroup->getRoleId(), [Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT])) {
-                        $userCanAccess = true;
-                        break;
-                    }
+        // submission preview will work for those who can see it.
+        if($applicationName == 'ojs2'){
+            if (!$this->issue || !$this->issue->getPublished() || $this->submission->getStatus() != PKPSubmission::STATUS_PUBLISHED) {
+                if (!$this->canUserAccess($context, $user)) {
+                    $request->getDispatcher()->handle404();
                 }
             }
+        }
+    }
 
-            if ($user && $user->hasRole(Role::ROLE_ID_MANAGER, $context->getId())) {
-                $userCanAccess = true;
+    private function canUserAccess($context, $user) {
```

Comment:
Nice, you can mark this `protected` though. We generally prefer `protected` over `private` so that someone can, if they want to, create a customized plugin based on this one.